plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -50,12 +53,31 @@ export async function createRouter(
     templaters,
     publishers,
     logger: parentLogger,
+    config,
     dockerClient,
   } = options;
 
   const logger = parentLogger.child({ plugin: 'scaffolder' });
   const jobProcessor = new JobProcessor();
 
+  const workingDirectory =
+    config.getOptionalString('scaffolder.workingDirectory') ?? os.tmpdir();
+  try {
+    // Check if working directory exists and is writable
+    await fs.promises.access(
+      workingDirectory,
+      fs.constants.F_OK | fs.constants.W_OK,
+    );
+    logger.info(`using working directory: ${workingDirectory}`);
+  } catch (err) {
+    logger.error(
+      `working directory ${workingDirectory} ${
+        err.code === 'ENOENT' ? 'does not exist' : 'is not writable'
+      }`,
+    );
+    throw err;
+  }
+
```

Comment:
I'm not really sure how to test this in the best way.

Reply:
You could use `mock-fs`, there's been another PR recently with some use cases #2926

Reply:
@benjdlambert thanks for the hint. I tried setting it up but keep getting the following error:
```
    Cannot find module './stream' from '../../../node_modules/winston/lib/winston/transports/index.js'



      at Resolver.resolveModule (../../../node_modules/jest-resolve/build/index.js:306:11)
      at Object.get [as Stream] (../../../node_modules/winston/lib/winston/transports/index.js:54:12)
      at Object.getVoidLogger (../../../packages/backend-common/src/logging/voidLogger.ts:45:45)
      at Object.<anonymous> (service/router.test.ts:51:17)
```
As soon as I remove my call to `mock-fs` that error goes away.

I'm not really sure what the problem is here so I'm hoping that you might have an idea. As far as I can see the files are there in my `node_modules` directory. I tried comparing with the setup from the issue that you linked but I can't see any significant differences. Not really sure how the `backstage-cli test` wires everything up.

Reply:
Hmm - not sure - i've not seen that error message before, maybe it's something to do with setting up a logger in the router, maybe try and mock some of that too to see if that error will go away. Looks like the logger is trying to require a file but mock-fs is jumping in and returning nothing